src/healthcheck.rs

```diff
@@ -0,0 +1,27 @@
+use std::error::Error;
+use std::net::SocketAddr;
+use tokio::io::AsyncWriteExt;
+use tokio::net::TcpListener;
+
+#[tracing::instrument]
+pub async fn build_healthcheck_server() -> Result<(), Box<dyn Error>> {
+    tracing::info!("Starting HTTP server");
+    let addr = SocketAddr::from(([127, 0, 0, 1], 3000));
+    let listener = TcpListener::bind(addr).await?;
+
+    loop {
+        let (mut socket, _) = listener.accept().await?;
+        tokio::spawn(async move {
+            match socket.peer_addr() {
+                Ok(peer_addr) => tracing::info!("Handling connection from {}", peer_addr),
+                Err(err) => tracing::info!("Handling connection from unknown peer {}", err),
```

Comment:
i think we use info too much, especially in this case. i think debug is most appropriate here

Reply:
Agree, but right now we have pinned the output to info (meaning debug won't be shown at all, even with the environment variable), so I'll change that in a separate pr